custom/lib/User.php

```diff
@@ -0,0 +1,211 @@
+            return $this->usersApi->getUser($this->appId, $userId)->getUser();
+        } catch (ApiException $e) {
+            throw PassageError::fromApiException($e);
+        }
+    }
+
+    /**
+     * Activate a user using their user ID.
+     *
+     * @param string $userId The Passage user ID
+     * @return PassageUser Passage User object
+     * @throws PassageError
+     */
+    public function activate(string $userId): PassageUser
+    {
+        try {
+            return $this->usersApi->activateUser($this->appId, $userId)->getUser();
+        } catch (ApiException $e) {
+            throw PassageError::fromApiException($e);
+        }
+    }
+
+    /**
+     * Deactivate a user using their user ID.
+     *
+     * @param string $userId The Passage user ID
+     * @return PassageUser Passage User object
+     * @throws PassageError
+     */
+    public function deactivate(string $userId): PassageUser
+    {
+        try {
+            return $this->usersApi->deactivateUser($this->appId, $userId)->getUser();
+        } catch (ApiException $e) {
+            throw PassageError::fromApiException($e);
+        }
+    }
+
+    /**
+     * Update a user.
+     *
+     * @param string $userId The Passage user ID
+     * @param UpdateUserArgs $args The updated user information
+     * @return PassageUser Passage User Object
+     * @throws PassageError
+     */
+    public function update(string $userId, UpdateUserArgs $args): PassageUser
+    {
+        try {
+            return $this->usersApi->updateUser($this->appId, $userId, $args)->getUser();
+        } catch (ApiException $e) {
+            throw PassageError::fromApiException($e);
+        }
+    }
+
+    /**
+     * Create a user.
+     *
+     * @param CreateUserArgs $args Arguments for creating a user
+     * @return PassageUser Passage User object
+     * @throws PassageError
+     */
+    public function create(CreateUserArgs $args): PassageUser
+    {
+        try {
+            return $this->usersApi->createUser($this->appId, $args)->getUser();
+        } catch (ApiException $e) {
+            throw PassageError::fromApiException($e);
+        }
+    }
+
+    /**
+     * Delete a user using their user ID.
+     *
+     * @param string $userId The Passage user ID used to delete the corresponding user.
+     * @return void
+     * @throws PassageError
+     */
+    public function delete(string $userId): void
+    {
+        try {
+            $this->usersApi->deleteUser($this->appId, $userId);
+        } catch (ApiException $e) {
+            throw PassageError::fromApiException($e);
+        }
+    }
+
+    /**
+     * Get a user's devices using their user ID.
+     *
+     * @param string $userId The Passage user ID
+     * @return WebAuthnDevices[] List of devices
+     * @throws PassageError
+     */
+    public function listDevices(string $userId): array
+    {
+        try {
+            return $this->userDevicesApi->listUserDevices($this->appId, $userId)->getDevices();
```

Comment:
Does this need to ensure `listUserDevices` was successful/non-null before `getDevices` is added? Or is this the expected syntax in PHP?

The error snapshot diff when testing against this branch looks like
```
    - Snapshot  - 4
    + Received  + 1

    - [400] Client error: `GET https://api.passage.id/v1/apps/***/users//devices` resulted in a `400 Bad Request` response:
    - Invalid format for parameter user_id: parameter 'user_id' is empty, can't bind its value
    -
    -
    + "Warning: Trying to access array offset on null"
```

which is a simpler error, but opaque as to what might've gone wrong.

Reply:
this is the expected syntax. the method would've errored at `$this->userDevicesApi->listUserDevices($this->appId, $userId)` before it hit the `->getDevices()` method

this is part of the blank id parameter guards that i can do across the sdks in a separate PR